File: data/src/main/java/com/example/data/repository/factory/InviteRepositoryFactoryImpl.kt
package com.example.data.repository.factory

import com.example.data.datasource.remote.InviteRemoteDataSource
import com.example.data.repository.base.InviteRepositoryImpl
import com.example.domain.repository.RepositoryFactory
import com.example.domain.repository.base.InviteRepository
import com.example.domain.repository.factory.context.InviteRepositoryFactoryContext
import javax.inject.Inject

class InviteRepositoryFactoryImpl @Inject constructor(
    private val inviteRemoteDataSource: InviteRemoteDataSource
) : RepositoryFactory<InviteRepositoryFactoryContext, InviteRepository> {

    override fun create(input: InviteRepositoryFactoryContext): InviteRepository {
        return InviteRepositoryImpl(
            inviteRemoteDataSource = inviteRemoteDataSource
        )
    }
}

File: functions/src/domain/invite/repositories/factory/InviteRepositoryFactoryImpl.ts
import { InviteRepositoryFactory } from './InviteRepositoryFactory';
import { InviteRepositoryFactoryContext } from './InviteRepositoryFactoryContext';
import { InviteRepository } from '../invite.repository';
import { InviteRepositoryImpl } from '../../../../infrastructure/repositories/invite.repository.impl';
import { FirestoreInviteDataSource } from '../../../../infrastructure/datasources/firestore/invite.datasource';

export class InviteRepositoryFactoryImpl implements InviteRepositoryFactory {
  create(context?: InviteRepositoryFactoryContext): InviteRepository {
    const dataSource = new FirestoreInviteDataSource();
    return new InviteRepositoryImpl(dataSource);
  }
}